tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -3877,3 +3875,425 @@ def test_complex128():
     assert (f_complex64() == test_complex64())
     assert (f_complex128() == test_complex128())
 
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = [
+            pytest.mark.skip(reason="Cimage is not supporting arrays yet"),
```

Comment:
To support `imag` for arrays you need to set `is_elemental` to true.

See `NumpyExp` for an example by following the inheritance tree
`NumpyExp`->`NumpyUfuncUnary`->`NumpyUfuncBase`->`PyccelInternalFunction`